src/devices/bus/spectrum/zxbus.h

```diff
@@ -107,15 +120,14 @@ DECLARE_DEVICE_TYPE(ZXBUS, zxbus_device)
 
 class device_zxbus_card_interface : public device_interface
 {
-public:
-	virtual ~device_zxbus_card_interface();
-
-	void set_zxbusbus(zxbus_device &bus) { assert(!device().started()); m_zxbus = &bus; }
+  friend class zxbus_slot_device;
 
 protected:
 	device_zxbus_card_interface(const machine_config &mconfig, device_t &device);
 
-	virtual void interface_pre_start() override;
+	void interface_pre_start() override;
+
+	void set_zxbusbus(zxbus_device *zxbus) { assert(!device().started()); m_zxbus = zxbus; }
```

Comment:
The bus should never be null - the parameter should be a reference rather than a pointer.